src/features/foldingProvider.ts

```diff
@@ -48,6 +51,34 @@ export default class AsciidocFoldingRangeProvider implements vscode.FoldingRange
     return listOfRanges
   }
 
+  private static getOpenBlockFoldingRanges (document: vscode.TextDocument) {
+    const listOfRanges = []
+    const openBlockIndexes = []
+    let end: number | undefined
+    for (let lineIndex = 0; lineIndex < document.lineCount; lineIndex++) {
+      const line = document.lineAt(lineIndex)
+      if (line.text === '--') {
+        openBlockIndexes.push(lineIndex)
+      }
+      if (line.text === '[source]') {
```

Comment:
The documentation shows two examples but here's a valid list of open blocks:

```
[.foo.bar]
--
text
--

[f]
--
text
--

[]
--
text
--

[#bar]
--
text
--

[opts=a]
--
text
--

[foo=bar]
--
text
--
```

As you can see, the only requirement is that it must start with `[` and it must end with `]` .